xonsh/environ.py

```diff
@@ -176,7 +176,7 @@ def DEFAULT_ENSURERS():
     'XONSH_DEBUG': (always_false, to_debug, bool_or_int_to_str),
     'XONSH_ENCODING': (is_string, ensure_string, ensure_string),
     'XONSH_ENCODING_ERRORS': (is_string, ensure_string, ensure_string),
-    'XONSH_HISTORY_BACKEND': (is_string, ensure_string, ensure_string),
+    'XONSH_HISTORY_BACKEND': (is_string_or_callable, ensure_string, ensure_string),
```

Comment:
Yes, that's what I mean, this `PROMPT` is like `XONSH_HISTORY_BACKEND` will be refresh to what user want them to be in xonshrc every time xonsh read xonshrc :) so no one would complain (found) assign a class type to it.